src/main/java/edu/greenblitz/robotName/commands/arm/roller/ShootFromRollerToAmp.java

```diff
@@ -0,0 +1,26 @@
+package edu.greenblitz.robotName.commands.arm.roller;
+
+import edu.greenblitz.robotName.subsystems.arm.EndEffector.RollerUtils.RollerConstants;
+import edu.greenblitz.robotName.subsystems.arm.Wrist;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+
+public class ShootFromRollerToAmp extends RollerCommand{
+    @Override
+    public void initialize() {
+        roller.rollIn();
+    }
+
+    @Override
+    public boolean isFinished() {
+        if (!Wrist.getInstance().isObjectInside()){
+            new WaitCommand(RollerConstants.TIME_AFTER_OBJECT_IN_ROBOT);
```

Comment:
wont work. consider adding .schedule.
but still, it won't work. the new WaitCommand is creating a new thread to run while other commands run so it won't stop the current command and wait 0.5 seconds. you can either do what @tnrz135 suggests or make a sequential command group.